exchanges/stream/websocket.go

```diff
@@ -1075,14 +1075,27 @@ func (w *Websocket) checkSubscriptions(conn Connection, subs subscription.List)
 // Reader reads and handles data from a specific connection
 func (w *Websocket) Reader(ctx context.Context, conn Connection, handler func(ctx context.Context, message []byte) error) {
 	defer w.Wg.Done()
+	var reporter ProcessReporter
+	if w.processReporter != nil {
+		reporter = w.processReporter.New(conn)
+	}
 	for {
 		resp := conn.ReadMessage()
+		readAt := time.Now()
```

Comment:
```suggestion
var readAt time.Time
if reporter != nil {
	readAt := time.Now()
}
```
`40.01 ns/op` vs `0.3179 ns/op` for those without a processReporter on a hot path

It is a shame gorilla doesn't have time received per message so we wouldn't have to call it.